dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/tracker/MetadataItemsHandler.java

```diff
@@ -208,7 +217,19 @@ private Map<String, List<String>> getDimensionItems(
     for (QueryItem item : params.getItems()) {
       String itemUid = getItemUid(item);
 
-      if (item.hasOptionSet()) {
+      if (item.getValueType().isOrganisationUnit()) {
+        dimensionItems.put(
+            itemUid,
+            item.getFilters().stream()
+                .map(
+                    queryFilter ->
```

Comment:
It's more readable if we move this invocation outside:
```
                        organisationUnitResolver.resolveOrgUnits(
                            queryFilter, params.getUserOrgUnits())
```

Reply:
it's a for cycle now, so most likely this is not needed anymore